gap/congruences/cong.gi

```diff
@@ -21,46 +21,49 @@
 ## cong.gd contains declarations for many of these.
 ##
 
-InstallMethod(\=, "for two semigroup congruences",
-[IsSemigroupCongruence, IsSemigroupCongruence],
-function(c1, c2)
-  if not IsFinite(Range(c1)) then
-    TryNextMethod();
+BindGlobal("_GenericCongEquality",
+function(cong1, cong2)
+  local S;
+  if Range(cong1) <> Range(cong2) then
+    return false;
   fi;
-  return Range(c1) = Range(c2)
-         and EquivalenceRelationCanonicalLookup(c1) =
-             EquivalenceRelationCanonicalLookup(c2);
+  S := Range(cong1);
```

Comment:
100% pedantic, I admit it, but you could put this line `S := Range(cong1)`; before the `if` statement, so that it then reads `if S <> Range(cong2) then`. It saves `Range(cong1)` having to be called twice, which I understand isn't a big overhead, but it would make me happier :).